website/pages/payments.html

```diff
@@ -68,16 +66,10 @@
     </tr>
     </thead>
       {{ for(var p in it.stats.pools[pool].payments) { }}
-<tr>
+      <tr>
           <td>
-          {{if (String(it.stats.pools[pool].name).startsWith("zcash")) { }}
-          <a href="https://explorer.zcha.in/tx/{{=it.stats.pools[pool].payments[p].txid}}"  title="View transaction" target="_blank">{{=it.stats.pools[pool].payments[p].blocks}}</a>
-          {{ } else if (String(it.stats.pools[pool].name).startsWith("zclassic")) { }}
-            <a href="https://classic.zcha.in/tx/{{=it.stats.pools[pool].payments[p].txid}}"  title="View transaction" target="_blank">{{=it.stats.pools[pool].payments[p].blocks}}</a>
-          {{ } else if (String(it.stats.pools[pool].name).startsWith("hush")) { }}
-            <a href="https://explorer.myhush.org/tx/{{=it.stats.pools[pool].payments[p].txid}}"  title="View transaction" target="_blank">{{=it.stats.pools[pool].payments[p].blocks}}</a>
-          {{ } else if (String(it.stats.pools[pool].name).startsWith("zen")) { }}
-            <a href="http://node1.zenchain.info:8886/tx/{{=it.stats.pools[pool].payments[p].txid}}"  title="View transaction" target="_blank">{{=it.stats.pools[pool].payments[p].blocks}}</a>
+          {{if (it.poolsConfigs[pool].explorer) { }}
+          <a href="https://{{=it.poolsConfigs[pool].explorer}}/tx/{{=it.stats.pools[pool].payments[p].txid}}" title="View transaction" target="_blank">{{=it.stats.pools[pool].payments[p].blocks}}</a>
```

Comment:
We should probably just use string interpolation, setting the config value to something like:

```
https://coin-domain.xyz/tx/%s
```

This will allow this change to work for current and new explorers which may not utilize the current expected url structure.

Reply:
Something like this?

```javascript
"explorerOptions: {
    "enabled": true,
    "secureExplorer": true,
    "txExplorer": "site.url.com/tx/",
    "blockExplorer": "siteurl.com/block/"
},
```

Reply:
Well, this is unfortunate. It doesn't look like nodejs supports native `sprintf` functionality. One thing you can do is something like:

In config, set `"explorer": "https://coin-domain.xyz/tx/%s",`

In the template, do `<a href="{{=it.poolsConfigs[pool].explorer.replace(/%s/, it.stats.pools[pool].payments[p].txid)"...`

Reply:
Hmm. What if we let a function handle the output? Untested

```
var explorerAddress = explorerLink("block", "12345");

//tx
	//explorerInput = it.stats.pools[pool].payments[p].txid
//block
	//explorerInput = block[0]

function explorerLink(linkType, explorerInput){

    //vars
    var enabled = it.poolsConfigs[pool].explorerOptions.enabled;
    var secureExplorer = it.poolsConfigs[pool].explorerOptions.secureExplorer;
    var txExplorer = it.poolsConfigs[pool].explorerOptions.txExplorer;
    var blockExplorer = it.poolsConfigs[pool].explorerOptions.blockExplorer;
    
    //Needed
    	//Strip http and https to let secureExplorer define it
    	//Add ending / if not given by user

    if(enabled) {
        if(secureExplorer) {
            //https://
            var http = "https://"
        } else {
            //http://
            var http = "http://"
        }
        
        if(linkType == 'tx') {
            //txExplorer
            var explorerType = txExplorer;
            
        } else if(linkType = 'block') {
            //blockExplorer
            var explorerType = blockExplorer;
        }
        
        //Return URL
        return http + explorerType + explorerInput;
        //https://coinexplorer.xyz/block/12345
    }
    
}
```

Reply:
There are a few issues with that approach:
- You need to ensure that function is available anywhere you need to output a link. This can be annoying if it's used in many places.
- It seems a bit over-engineered.

I do agree that we need more than just `explorer` being a string. The enabled and secure seem a bit redundant since you probably wouldn't disable an explorer -- but the whole coin. When you do that, you wouldn't print anything for the explorer since the coin won't appear in the loop.

Reply:
Check my latest commits. I think this will be the most straightforward and easy way to complete this without over-engineering it. I agree the function idea is a bit over the top. 

```
"explorerOptions": {
    "txExplorer": "https://site.url.com/tx/",
    "blockExplorer": "https://site.url.com/block/"
},
```
If you approve, I can udpate the explorerOptions to the example configs.